package/gluon-core/luasrc/lib/gluon/upgrade/115-swconfig

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/lua
+
+local platform = require 'gluon.platform'
+local uci = require('simple-uci').cursor()
+
+local function get_keys(t)
+	local keys = {}
+	local n = 0
+
+	for k,_ in pairs(t) do
+		n = n+1
+		keys[n] = k
+	end
+	return keys
+end
+
+local switch_vlans = {
+	-- device identifier, lan ports, wan ports
+	["ubnt,unifiac-pro"] = {"2 0t", "3 0t"},
+}
+
+local devices = get_keys(switch_vlans)
+
+if not platform.match('ath79', 'generic', devices) then
```

Comment:
Actually, we don't even need the `get_keys()` helper:
```suggestion
local board_name = platform.get_board_name()
local assignments = switch_vlans[board_name]
if not platform.match('ath79', 'generic') or not assignments then
```

Reply:
resolved in 4fbc4e0cd17fb40269dc39f0525f9cab06c0e11c